Celltype report: Add introduction, sessioninfo, and include child celltype_qc (#539)
AlexsLemonade/scpca-nf: templates/qc_report/celltypes_qc.rmd

<!-- Call but do not evaluate text --> 
```{r child = 'celltypes_introduction.rmd'}

> ```

Reviewer: I actually meant to put this into the main report right above the celltype report. Also, nextflow will complain about a nested child report.

# Cell type Annotation Summary

```{r methods-info-box, eval = !is_supplemental, results = "asis"}
```

> 
> 
> 

Author: so many lines!
```suggestion
```

```

The plots and tables included here detail the results from performing cell type annotation.

```{r, eval = !is_supplemental, results = 'asis'}
# define bullets glue string
methods_bullets <- c(
  ifelse(has_submitter, "Submitter-provided", NA),
  ifelse(has_singler, "`SingleR`", NA),
  ifelse(has_cellassign, "`CellAssign`", NA)
) |>
  na.omit() |>
  # make it bullets
  stringr::str_replace_all("^", "* ") |>
  stringr::str_flatten(collapse = "\n")

# print the bullets
glue::glue("
  This library contains the following cell type annotations:
    
  {methods_bullets}
  

>   For additional information about cell typing, including methodologies and information about reference sources, comparisons among cell type annotation methods, and diagnostic plots, please refer to the [supplementary cell type QC report](`r params$celltype_report`).

Reviewer: ```suggestion
  For additional information about cell typing, including methods used for cell typing, information about reference sources, comparisons among cell type annotation methods, and diagnostic plots, please refer to the [supplementary cell type QC report](`r params$celltype_report`).
```
You don't need to accept this, but I think this is clearer about what you mean by methodologies.

Author: agreed!

<!-- Background depends on which cell types are present in the SCE. -->

The plots and tables included here detail the results from performing cell type annotation.
Cell type annotations were obtained as follows: 

```{r results='asis'}
# build up this string as we go:
methods_text <- ""

if (has_submitter) {
  methods_text <- glue::glue(
    "{methods_text}

>     * Cell type annotations were provided by the original lab (`submitter-provided`) which generated this data.\n\n"

Reviewer 2: Given that `glue` is adding a linebreak when you next append, I think you have an extra here?

Also reworded for simplicity and removing code style.
```suggestion
    * Submitter-provided cell type annotations were generated by the original lab which produced this data.\n"
```

Author: >  I think you have an extra here?

Yes, I added it intentionally for legibility. But, it's not needed for a "shorter and sweeter" box.

  )
}

if (has_singler) {
  
  methods_text <- glue::glue(
    "{methods_text}
    * Cell types were annotated using [`SingleR`](https://bioconductor.org/packages/release/bioc/html/SingleR.html), which uses a reference-based approach ([Looney _et al._ 2019](https://doi.org/10.1038/s41590-018-0276-y)).
    The `{metadata(processed_sce)$singler_reference}` reference dataset, obtained from the [`celldex` package](http://bioconductor.org/packages/release/data/experiment/html/celldex.html), was used for cell typing.\n\n"
    )
}

if (has_cellassign) {

  # TODO: UPDATE THIS STRING WRANGLING WHEN NAMING CHANGES???
  tissue_type <- stringr::word(
    metadata(processed_sce)$cellassign_reference,
    -1,
    sep = "-"
  )
  
  methods_text <- glue::glue(
    "{methods_text}
    * Cell types were annotated using [`CellAssign`](https://github.com/Irrationone/cellassign), which uses a marker-gene based approach ([Zhang _et al._ 2019](https://doi.org/10.1038/s41592-019-0529-1)).

>     Marker genes associated with `{tissue_type}` tissue were obtained from [PanglaoDB](https://panglaodb.se/).\n

Reviewer 2: The wording here may need also an update based on changes in the way we name the PangaloDB cell sets. We probably also want some kind of link to how we define those cell sets...

Which leads me to a general comment that I think these bullets are too long for an alert box. This kind of detail makes sense in the supplemental report, but I don't think we need it at the top, or in the main report. I would more compactly name the methods here. (Both for CellAssign and SingleR)

Then I would include this detailed information later in the supplemental report only.

    "
  )
}

glue::glue("
<div class=\"alert alert-info\">

{methods_text}

</div>
") 
```

<!-- At least one cell type annotation method required for the supplemental report -->
```{r, eval = !is_supplemental & (has_singler | has_cellassign), results='asis'}
glue::glue("

>   For additional information about cell typing, including diagnostic plots and heatmap comparisons across cell labels, please refer to the [supplementary cell type QC report](`r params$celltype_report`).

Reviewer 2: Maybe something like this?
```suggestion
  For additional information about cell typing, including details on reference sources, comparisons among cell type annotation methods, and diagnostic plots, please refer to the [supplementary cell type QC report](`r params$celltype_report`).
```

<!-- Background depends on which cell types are present in the SCE. -->

The plots and tables included here detail the results from performing cell type annotation.
Cell type annotations were obtained as follows: 

<!-- Call chunk to print out text --> 

> ```{r available_celltypes_text, results='asis', eval=TRUE}

Reviewer: I think if you remove the `eval = FALSE` and add `results='asis'` to `celltype_introduction.rmd`, you should just be able to include the intro as a child report and won't need to do this?
